File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */



const longestConsecutive = nums => {
    if (nums.length === 1) return 1;
    
    const set = new Set();
    let longestCount = 0;
    
    for (let i = 0; i < nums.length; i++) {
        set.add(nums[i]);
    }
    
    for (let i = 0; i < nums.length; i++) {
        let num = nums[i];
        if (!set.has(num - 1)) {
            let currentCount = 1;
            while (set.has(num + 1)) {
                num++;
                currentCount++;
            }
            longestCount = Math.max(currentCount, longestCount);
        }
    }
    
    return longestCount;
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        num_set = set(nums)
        longest_seq = 0
        for i in range(len(nums)):
            num = nums[i]
            if num - 1 not in num_set:
                curr_seq = 1
                while num + 1 in num_set:
                    curr_seq += 1
                    num += 1
                longest_seq = max(longest_seq, curr_seq)
        return longest_seq
                